sdk/identity/src/token_credentials/specific_azure_credential.rs

```diff
@@ -0,0 +1,340 @@
+    AzureCli(AzureCliCredential),
+    VirtualMachine(VirtualMachineManagedIdentityCredential),
+    AppService(AppServiceManagedIdentityCredential),
+    ClientSecret(ClientSecretCredential),
+    WorkloadIdentity(WorkloadIdentityCredential),
+    #[cfg(feature = "client_certificate")]
+    ClientCertificate(ClientCertificateCredential),
+}
+
+#[cfg_attr(target_arch = "wasm32", async_trait::async_trait(?Send))]
+#[cfg_attr(not(target_arch = "wasm32"), async_trait::async_trait)]
+impl TokenCredential for SpecificAzureCredentialEnum {
+    async fn get_token(&self, scopes: &[&str]) -> azure_core::Result<AccessToken> {
+        match self {
+            SpecificAzureCredentialEnum::Environment(credential) => {
+                credential.get_token(scopes).await
+            }
+            #[cfg(not(target_arch = "wasm32"))]
+            SpecificAzureCredentialEnum::AzureCli(credential) => credential.get_token(scopes).await,
+            SpecificAzureCredentialEnum::VirtualMachine(credential) => {
+                credential.get_token(scopes).await
+            }
+            SpecificAzureCredentialEnum::AppService(credential) => {
+                credential.get_token(scopes).await
+            }
+            SpecificAzureCredentialEnum::ClientSecret(credential) => {
+                credential.get_token(scopes).await
+            }
+            SpecificAzureCredentialEnum::WorkloadIdentity(credential) => {
+                credential.get_token(scopes).await
+            }
+            #[cfg(feature = "client_certificate")]
+            SpecificAzureCredentialEnum::ClientCertificate(credential) => {
+                credential.get_token(scopes).await
+            }
+        }
+    }
+
+    async fn clear_cache(&self) -> azure_core::Result<()> {
+        match self {
+            SpecificAzureCredentialEnum::Environment(credential) => credential.clear_cache().await,
+            #[cfg(not(target_arch = "wasm32"))]
+            SpecificAzureCredentialEnum::AzureCli(credential) => credential.clear_cache().await,
+            SpecificAzureCredentialEnum::VirtualMachine(credential) => {
+                credential.clear_cache().await
+            }
+            SpecificAzureCredentialEnum::AppService(credential) => credential.clear_cache().await,
+            SpecificAzureCredentialEnum::ClientSecret(credential) => credential.clear_cache().await,
+            SpecificAzureCredentialEnum::WorkloadIdentity(credential) => {
+                credential.clear_cache().await
+            }
+            #[cfg(feature = "client_certificate")]
+            SpecificAzureCredentialEnum::ClientCertificate(credential) => {
+                credential.clear_cache().await
+            }
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct SpecificAzureCredential {
+    source: SpecificAzureCredentialEnum,
+}
+
+impl SpecificAzureCredential {
+    pub fn create(options: TokenCredentialOptions) -> azure_core::Result<SpecificAzureCredential> {
```

Comment:
Come to think of it, why `create` vs. `new`?

Reply:
For the credentials, `new` is being used when you are passing everything in. `create` is being used  you are trying to create the credential from environment variables or via a filesystem check or whatnot in TokenCredentialOptions. It is similar to `TryCreate` that .NET is using.